### PR TITLE
🎉 New Source: Aircall [Low code CDK]

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Developing new connector for the source: Aircall*

Resolves:
[Aircall](https://github.com/airbytehq/DeveloperWeek-Connector-Contest/issues/11)

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## Tests

<details open><summary><strong>Integration & Acceptance</strong></summary>
Full Test Results

image

</details>

## 🚨 User Impact 🚨
- No breaking changes, Just addition of new source
<details><summary><strong>New Connector: Source Yotpo</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by btkcodedev** [comment]
@RealChrisSean, @marcosmarxm 
PR raised for review
All acceptance tests are passing
Please ping me via slack for secret sharing

**Comment by btkcodedev** [comment]
@marcosmarxm @RealChrisSean, Unfortunately, I've got only 7-day trial with aircall secrets.
I would greatly appreciate the testing as soon as possible :)
I would do my best for resolving comments

File: airbyte-integrations/connectors/source-aircall/source_aircall/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      title: API ID
+      type: string
+      description: App ID found at settings https://dashboard.aircall.io/integrations/api-keys
+      airbyte_secret: true
+    api_token:
+      title: API Token
+      type: string
+      description: App token found at settings (Ref- https://dashboard.aircall.io/integrations/api-keys)
+      airbyte_secret: true
+    date_from:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aircall/source_aircall/spec.yaml:22]:
```suggestion
    start_date:
```

File: airbyte-integrations/connectors/source-aircall/source_aircall/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Aircall Spec
+  type: object
+  required:
+    - api_id
+    - api_token
+    - date_from
+  additionalProperties: true
+  properties:
+    api_id:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aircall/source_aircall/spec.yaml:12]:
```suggestion
    app_id:
```

File: airbyte-integrations/connectors/source-aircall/source_aircall/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+  company_stream:
+    type: DeclarativeStream
+    retriever:
+      type: SimpleRetriever
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path: ["company"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aircall/source_aircall/manifest.yaml:38]: Please create a base RecordSelector and use `field_path: ["{{ parameters['name'] }}"]`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Add some comments.

File: airbyte-integrations/connectors/source-aircall/source_aircall/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+      paginator:
+        type: NoPagination
```
[the remark this answers is not in the document]
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-aircall/source_aircall/manifest.yaml:40]: I have a doubt regarding pagination, 
the `page` and `per_page` is inside {meta: {}}, but the data is extracted from next field, 
Could you suggest a method for pagination

[the remark this answers is not in the document]
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-aircall/source_aircall/manifest.yaml:41]: @marcosmarxm

**Comment by btkcodedev** [comment]
Comments resolved and test passed

image

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**